All Core Devs Meetings/Meeting 102.md

```diff
@@ -0,0 +1,184 @@
 a full block or empty block. And it's just a linear increase and decrease based on the gas used. And he says that this is maybe not like the optimal way to do this, given that once you if you have a very quick increase, you might then need to have a bunch of empty blocks to lower the base fee enough to reach the price that people are willing to pay. So it's not the end of the world, but it's that slightly inefficient. So we have someone who's looking into possibly just a better formula for that to have a smoother update. And yeah, so I guess I just recommend anybody who kind of cares about 1559 and its impact on the network to at least have a skim the report that was quite good in terms of other progress. We've been working we've been working on the implementation a lot. So there's a there's a draft PR against the geth code base that's that's up. The implementation still don't have EIP 2718. So none of them are kind of ready for, you know, being merged and trying to master. But they're there interop together. We have a small test net. So that's that's a good sign. And we've kind of held off implementing 2017 until it was it was done and clients were brought in. And I guess the last outstanding thing on the client side that we need to figure out is the transaction pool management. So because the baozi changes from block to block, like the naïf sorting of transactions in the transaction pool, if you take a naive approach, you can end up resorting to transaction pool every block, and that's a pretty bad outcome. So we're working to find some better heuristics about how we can efficiently manage transactions. And the Quil team is helping out with that. So I guess, yeah, at a very high level, I suspect in another month or two we should have, like most of these issues, figured out that the implementations ironed out with 2718, then there'll be a bunch of there's a bunch of like auxiliary work like updating the Jason RBC suspects and whatnot. But the, the bulk I think of the 1559 implementations should be done and one. Yeah. Sorry. One last thing we're working on is large datasets. So because one of the concerns was how would clients on Main Net handle fifteen fifty nine one part of that is like the database has a pretty huge stap it becomes harder to process, it might become harder to process blocks that are twice as big. So Abdel from our team has put together I've got a large state test generator which can generate a new network with an arbitrary number of accounts and storage slots in the contract. So we're we're setting up a new desktop now with one hundred million accounts and a contract with one hundred million storage slots, which is more or less what we have a mainnet. And then once we have all the clients kind of running on that network, we'll start with a bunch of 1559 transactions and see if the clients can handle the load for like half an hour an hour, which is the max upper bound of what you'd expect on mainnet. And that's pretty much it.
```

Comment:
```suggestion
**Tim** - Yeah, I won't try to summarize too much of it here, but we had another 1559 call last week, I believe, and the big, I guess, take away or new thing was Tim Roughgarden who's a computer science and game theory professor from Columbia, released a paper analyzing 1559 from an economic perspective and trying to understand is it actually better than what we have today? I just shared the paper in the chat. It's pretty extensive and the 10 takeaways are shared in the third page. If you don't want to skim the 60 page paper, you can just read that. And overall, the paper is pretty positive, about 1559, which is nice. So just in terms of UX, he makes it clear that having the base fee as part of the network really helps kind of people converge on the optimal bids for their transactions. And that's a good thing. And that the incentives for miners to carry out the protocol as intended are the same under 1559 as under our current mechanism. The one I guess or sorry the two kind of more negative sections of the report was one around miners, where he highlights that today. There's a lot of negative behavior we could see from miners on the network that would theoretically be possible, but that doesn't happen and we don't have a good explanation for that. Aside from maybe the miners are eth holders or they care about their health of the networks that are somewhat altruistic. But he says, you know, that could potentially change under fifty nine. So if if for some reason miners decide to engage in behavior that today they could but don't pick up, that's a potential risk. And then the other this is a much more minor comment about it is the way that the update rule works. In 1559 we increase or lower the base fee by up to twelve and a half percent if we have a full block or empty block. And it's just a linear increase and decrease based on the gas used. And he says that this is maybe not like the optimal way to do this, given that once you if you have a very quick increase, you might then need to have a bunch of empty blocks to lower the base fee enough to reach the price that people are willing to pay. So it's not the end of the world, but it's that slightly inefficient. So we have someone who's looking into possibly just a better formula for that to have a smoother update. And yeah, so I guess I just recommend anybody who kind of cares about 1559 and its impact on the network to at least have a skim the report that was quite good in terms of other progress. We've been working we've been working on the implementation a lot. So there's a there's a draft PR against the geth code base that's that's up. The implementation still don't have EIP 2718. So none of them are kind of ready for, you know, being merged and trying to master. But they're there interop together. We have a small test net. So that's that's a good sign. And we've kind of held off implementing 2017 until it was it was done and clients were brought in. And I guess the last outstanding thing on the client side that we need to figure out is the transaction pool management. So because the Besu changes from block to block, like the naïf sorting of transactions in the transaction pool, if you take a naive approach, you can end up resorting to transaction pool every block, and that's a pretty bad outcome. So we're working to find some better heuristics about how we can efficiently manage transactions. And the Quil team is helping out with that. So I guess, yeah, at a very high level, I suspect in another month or two we should have, like most of these issues, figured out that the implementations ironed out with 2718, then there'll be a bunch of there's a bunch of like auxiliary work like updating the Jason RBC suspects and whatnot. But the, the bulk I think of the 1559 implementations should be done and one. Yeah. Sorry. One last thing we're working on is large datasets. So because one of the concerns was how would clients on Main Net handle fifteen fifty nine one part of that is like the database has a pretty huge stap it becomes harder to process, it might become harder to process blocks that are twice as big. So Abdel from our team has put together I've got a large state test generator which can generate a new network with an arbitrary number of accounts and storage slots in the contract. So we're we're setting up a new desktop now with one hundred million accounts and a contract with one hundred million storage slots, which is more or less what we have a mainnet. And then once we have all the clients kind of running on that network, we'll start with a bunch of 1559 transactions and see if the clients can handle the load for like half an hour an hour, which is the max upper bound of what you'd expect on mainnet. And that's pretty much it.
```